utils/collection/conditions.go

```diff
@@ -0,0 +1,213 @@
+	return c
+}
+
+// Add adds conditions and returns itself.
+func (c *Conditions) Add(conditions ...bool) Conditions {
+	if c == nil {
+		return nil
+	}
+	*c = append(*c, conditions...)
+	return *c
+}
+
+// ForEach will execute function each() on every condition unless an error is returned and will end add this point.
+func (c *Conditions) ForEach(each func(bool) error) error {
+	if c == nil || len(*c) == 0 {
+		return fmt.Errorf("%w: the collection of conditions is empty", commonerrors.ErrUndefined)
+	}
+	for i := range *c {
+		subErr := each((*c)[i])
+		if subErr != nil {
+			return subErr
+		}
+	}
+	return nil
+}
+
+// Contains returns whether the conditions collection contains the value of a condition.
+func (c *Conditions) Contains(condition bool) bool {
+	if c == nil {
+		return false
+	}
+	if condition {
+		return c.Any()
+	} else {
+		return !c.All()
+	}
+}
+
+// All returns true if all conditions are true.
+func (c *Conditions) All() bool {
+	if c == nil {
+		return false
+	}
+	return All(*c)
+}
+
+// Any returns true if  there is at least one condition which is true.
+func (c *Conditions) Any() bool {
+	if c == nil {
+		return false
+	}
+	return Any(*c)
+}
+
+// Concat concatenates conditions and returns itself.
+func (c *Conditions) Concat(more *Conditions) Conditions {
+	if more == nil {
+		return nil
+	}
+	return c.Add(*more...)
+}
+
+// Negate returns a new set of conditions with negated values.
+func (c *Conditions) Negate() Conditions {
+	if c == nil {
+		return nil
+	}
+	negation := Conditions(Negate(*c...))
+	return negation
+}
+
+// And performs an `and` operation on all conditions
+func (c *Conditions) And() bool {
+	if c == nil {
+		return false
+	}
+	return And(*c...)
+}
+
+// Or performs an `Or` operation on all conditions
+func (c *Conditions) Or() bool {
+	if c == nil {
+		return false
+	}
+	return Or(*c...)
+}
+
+// Xor performs a `Xor` operation on all conditions
+func (c *Conditions) Xor() bool {
+	if c == nil {
+		return false
+	}
+	return Xor(*c...)
+}
+
+// OneHot performs an `OnHot` operation on all conditions
+func (c *Conditions) OneHot() bool {
+	if c == nil {
+		return false
+	}
+	return OneHot(*c...)
+}
+
+// Any returns true if there is at least one element of the slice which is true.
+func Any(slice []bool) bool {
+	if len(slice) == 0 {
+		return false
+	}
+	for i := range slice {
+		if slice[i] {
+			return true
+		}
+	}
+	return false
+}
+
+// AnyTrue returns whether there is a value set to true
+func AnyTrue(values ...bool) bool {
+	return Any(values)
+}
+
+// All returns true if all items of the slice are true.
+func All(slice []bool) bool {
+	if len(slice) == 0 {
+		return false
+	}
+	for i := range slice {
+		if !slice[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// AllTrue returns whether all values are true.
+func AllTrue(values ...bool) bool {
+	return All(values)
+}
+
+// Negate returns the slice with contrary values.
+func Negate(values ...bool) []bool {
+	if len(values) == 0 {
+		return nil
```

Comment:
i think that a nil slice is technically different from an empty slice so if the slice is empty but not nil then it should return `[]bool{}` or `make([]bool, 0)

Reply:
I addressed this